include/app.hrl

```diff
@@ -16,3 +16,7 @@
         {keepalive,true}, % Keep our channel open
         {tos,72}, % Deliver immediately
         {active,false}]). % Retrieve data from socket upon request
+
+-type socket_setopt() ::
```

Comment:
I see your point but I think there's no need to define a type, we can directly use the gen_tcp:option() spec since we're only using that.